cds/modules/deposit/indexer.py

```diff
@@ -43,3 +47,52 @@ def cdsdeposit_indexer_receiver(
     if record['$schema'] in [project_schema, video_schema]:
         json['_cds']['state'] = deposit['_cds']['state']
         json['_files'] = deposit['_files']
+
+
+class CDSRecordIndexer(RecordIndexer):
+
+    def index(self, deposit, action='commit'):
+        project_schema = current_jsonschemas.path_to_url(Project._schema)
+        if action == 'publish':
+            if deposit['$schema'] == project_schema:
+                # index videos (records)
+                pid_values = Project(data=deposit).video_ids
+                ids = [
```

Comment:
When you publish a project, the `publish` action is called on each video (https://github.com/CERNDocumentServer/cds-videos/blob/cdslabs_qa/cds/modules/deposit/api.py#L542).
I think this is not necessary, and  it will re-index 2 times the videos, or maybe I am wrong.

You can test if this code is executed when you publish the project, by checking if this code is called for every video and for the project.

Reply:
This part of code is called when we publish through the api like here https://github.com/zzacharo/cds-videos/blob/565d18670094fced4e379fbfe58710b27279a6ac/tests/unit/test_project_rest.py#L208 . When we explicitly call the publish method we should manually index after.